salt/modules/logmod.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+'''
+On-demand logging
+=================
+
+.. versionadded:: Nitrogen
+
+The sole purpose of this module is logging messages in the (proxy) minion.
+It comes very handy when debugging complex Jinja templates, for example:
+
+.. code-block:: jinja
+
+    {%- for var in range(10) %}
+      {%- set logged = salt.log.info(var) %}
+    {%- endfor %}
+
+CLI Example:
+
+.. code-block:: bash
+
+    salt '*' log.error 'Please dont do that, this module is not for CLI use!'
```

Comment:
I am going to say that I really like the idea to be able to send info into the logs via a remote command as well as embedded in states. This allows users to add log info on the fly to commands and really any salt operation.
As for log pollution, should we prepend something to look for so these could be filtered out? Or is the module that we are logging from enough (since it is already there)?